### DISCUSSION TITLE
How can I authenticate using prismaclient?

### DISCUSSION THREAD

**Discussion by yuripramos** [prisma/prisma#1917] [Q&A] [answered]
Hi Everyone I've been following this tutorial here: https://github.com/prisma/prisma-examples/tree/master/typescript/graphql-nextjs to bootstrap things and using the SignupMutation to create some new user.

But the question is how can I **authenticate** the user using Prisma client?

I've tried this tutorial before but https://www.prisma.io/tutorials/authentication-in-apollo-server-ct21

But this tutorial needs to set up your own database and I don't want to, that's why I've started to use Prisma client, they provide to me one DB inside the file `/prisma/dev.db`

First time using Prisma and would like some help, thanks, everybody!

**Comment by 2color** [comment, marked as answer]
Hey @yuripramos 👋🏼,

I suggest you look at the [GraphQL example with authentication](https://github.com/prisma/prisma-examples/tree/master/typescript/graphql-auth).

To implement authentication with Prisma, you generally use Prisma to store and get permissions from the database. The actual authentication logic is either handled by you in the request handler, the framework, or a plugin like https://github.com/maticzav/graphql-shield.

**Reply by yuripramos** [reply]
Thanks for your fast reply, I've been in rush these days I couldn't reply fast. I'm bootstrapping my project from this repo that you showed to me but when I try to use next in my client they change my `module:commonjs` to `esnext `in my `tsconfig.ts `  and my `server.ts` couldn't work properly because esnext are not understanding dynamic imports.

I got this error:


```

C:\projects\prisma-examples\typescript\graphql-auth\src\server.ts:1   

import { GraphQLServer } from 'graphql-yoga';                                                                                              

^                                                                                                                                                                         

SyntaxError: Unexpected token {


```


Any tips? I'm looking for a workaround but without success =(

**Comment by yuripramos** [comment]
I've solved this problem creating 2 tsconfig files one` tsconfig.ts `and` tsconfig.server.json` and extended from `tsconfig.ts` and set up `module:commonjs` inside my server file.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
